plugins/Jobs.cpp

```diff
@@ -418,13 +427,25 @@ bool BedrockPlugin_Jobs::processCommand(BedrockNode* node, SQLite& db, BedrockNo
         const string& nextRun = result[0][1];
         const string& lastRun = result[0][2];
         string repeat = result[0][3];
+        const int parentJobID = SToInt(result[0][4]);
 
         // Make sure we're finishing a job that's actually running
         if (state != "RUNNING") {
             SWARN("Trying to finish job#" << request["jobID"] << ", but isn't RUNNING (" << state << ")");
             throw "405 Can only retry/finish RUNNING jobs";
         }
 
+        // If we are finishing a job that has child jobs, set its state to paused.
+        const int childCount = _getCountPendingChildJobs(db, request.calc64("jobID"));
```

Comment:
You can save a query by moving this call into the `if(SIEquals(request.methodLine, "FinishJob"))`, nab  #prematureoptimization